### PR TITLE
Update examples to support apple silicon GPU acceleration

### PR DESCRIPTION
**Pull request by NripeshN**

### CODE REVIEW

**Review by rusty1s — COMMENTED** [review]
Thanks for the PR. This is a massive one. How would we know that all of them work with `MPS` backend? Can we start more lightweight here and just add this to a few of the main ones?

**Comment by NripeshN** [comment]
> Thanks for the PR. This is a massive one. How would we know that all of them work with `MPS` backend? Can we start more lightweight here and just add this to a few of the main ones?

Hi, sure I'll revert back the changes and add only this to the main ones. Could I please know what are the main examples?

**Comment by NripeshN** [comment]
> Thanks for the PR. This is a massive one. How would we know that all of them work with `MPS` backend? Can we start more lightweight here and just add this to a few of the main ones?

I have tested every example one by one on MPS background on my apple silicon mac. Everything works on GPU Acceleration

File: examples/multi_gpu/data_parallel.py [hunk 1]
```diff
@@ -53,7 +53,12 @@ def forward(self, data):
 model = Net()
 print(f"Let's use {torch.cuda.device_count()} GPUs!")
 model = DataParallel(model)
-device = torch.device('cuda:0' if torch.cuda.is_available() else 'cpu')
+if torch.cuda.is_available():
+    device = torch.device('cuda:0')
+elif hasattr(torch.backends, 'mps') and torch.backends.mps.is_available():
+    device = torch.device('mps')
+else:
+    device = torch.device('cpu')
```
Comment (akihironitta) [inline examples/multi_gpu/data_parallel.py:61]: > I have tested every example one by one on MPS background on my apple silicon mac. Everything works on GPU Acceleration

@NripeshN Are you sure everything works? Data Parallel doesn't work on Apple Silicon afaik.

Comment (NripeshN) [reply examples/multi_gpu/data_parallel.py:61]: Hi, it worked fine on my laptop without throwing any errors but this program did not run on my friend's laptop. I am not entirely sure why.

Comment (NripeshN) [reply examples/multi_gpu/data_parallel.py:61]: @akihironitta @rusty1s I am sorry for my comment earlier, I had written a script that would run all Python scripts in a directory and print out the files that throw out errors. Unfortunately that program issues and showed everything passed.

**Comment by rusty1s** [comment]
Hey, can we still add support for this in the main examples, e.g., `gcn.py`, `gat.py`, `mutag_gin.py`, `link_pred.py`, `ppi.py`. I am also curious how this behaves with `NeighborLoader`, e.g., in the `reddit.py` example. Might be good to confirm in a separate PR.

**Pull request closed without merging**